multiapi-engine/src/main/java/com/sngular/api/generator/plugin/asyncapi/AsyncApiGenerator.java

```diff
@@ -160,6 +153,36 @@ public final void processFileSpec(final List<SpecFile> specsListFile) {
     }
   }
 
+  private void handleMissingPublisherConsumer(final SpecFile fileParameter, final JsonNode channel, final String operationId) {
+    if (channel.has(SUBSCRIBE) && ObjectUtils.allNull(fileParameter.getConsumer(), fileParameter.getStreamBridge())) {
```

Comment:
Lines 157-169 and L.171-182 are very similar.
May be we can made a refactor for common code, the first thing that I see:
```
var parameter = OperationParameterObject.builder()
                                    .ids(operationId)
                                    .apiPackage(DEFAULT_ASYNCAPI_API_PACKAGE)
                                    .modelPackage(DEFAULT_ASYNCAPI_MODEL_PACKAGE)
                                    .build());
...
     try {
        checkClassPackageDuplicate(CONSUMER_CLASS_NAME, DEFAULT_ASYNCAPI_API_PACKAGE);
        fileParameter.setConsumer(parameter);
      } catch (final DuplicateClassException ignored) {
        // Don't set consumer
      }
    }
...
fileParameter.setSupplier(parameter);
...
```